packages/2d/src/Hooks/useInspectorSelectEntity.ts

```diff
@@ -0,0 +1,71 @@
+import { useNewComponent, useEntity } from "@hex-engine/core";
+import { useInspectorSelect } from "@hex-engine/inspector";
+import Geometry from "../Components/Geometry";
+import { useDraw } from "../Hooks";
+import Mouse from "../Components/Mouse";
+
+/**
+ * Sets up the Inspector so that if the Inspector is in select mode,
+ * any Entity that has a Geometry component and is hovered in world space
+ * will be highlighted. Clicks on highlighted Entities will report them to
+ * the Inspector.
+ *
+ * This function does nothing in release builds.
+ */
+export default function useInspectorHoverOutline(
+  getGeometry: () => ReturnType<typeof Geometry>
+) {
+  if (process.env.NODE_ENV === "production") return;
+
+  const entity = useEntity();
+  const geometry = getGeometry();
+
+  if (geometry === undefined) {
+    return;
+  }
+
+  const { getSelectMode, inspectEntity } = useInspectorSelect();
+
+  const { onEnter, onLeave, onClick } = useNewComponent(() =>
+    Mouse({ entity, geometry })
+  );
+
+  let hovered = false;
+
+  onEnter(() => {
+    if (getSelectMode()) {
+      hovered = true;
+    }
+  });
+
+  onLeave(() => {
+    if (getSelectMode()) {
+      hovered = false;
+    }
+  });
+
+  onClick(() => {
+    if (getSelectMode()) {
+      hovered = false;
+      inspectEntity(entity);
+    }
+  });
+
+  useDraw((context) => {
+    if (hovered) {
+      const shape = geometry!.shape;
+      const color = "#B076C7";
+      const alpha = 0.35;
+      const previousAlpha = context.globalAlpha;
+
+      context.fillStyle = color;
+      context.globalAlpha = alpha;
+      shape.draw(context, "fill");
+      context.globalAlpha = previousAlpha;
```

Comment:
Was this leaking into other stuff, or are you being precautious? We call `context.save()` and `context.restore()` as part of calling the stuff passed into `useDraw`, so unless `context.save()` and `context.restore()` doesn't handle globalAlpha, we shouldn't have to worry about the previous value... but if that's the case, we should probably reset it in `useDraw` instead, so people don't have to worry about it.

https://github.com/suchipi/hex-engine/blob/e80f316f18734052733f02755e6a500ee5b4dc0f/packages/2d/src/Hooks/useDraw.ts#L39

Reply:
No it wasn't, but I'm drawing an outline after that. I guess I could draw the outline first.

Reply:
ohh, I see. I misunderstood; no problem here, then